superset/views/core.py

```diff
@@ -2831,11 +2819,7 @@ def slice_query(self, slice_id):
         get the database query string for this slice
         """
         viz_obj = self.get_viz(slice_id)
-        if not security_manager.datasource_access(viz_obj.datasource):
-            return json_error_response(
-                security_manager.get_datasource_access_error_msg(viz_obj.datasource),
-                status=401,
-                link=security_manager.get_datasource_access_link(viz_obj.datasource))
+        self.check_datasource_permission(viz_obj.datasource)
```

Comment:
Same as mentioned above, this looks like it assumes that check_datasource_permission raises

Reply:
ditto